Libraries/Text/Text/RCTTextView.m

```diff
@@ -173,10 +252,14 @@ - (void)setTextStorage:(NSTextStorage *)textStorage
 - (void)drawRect:(CGRect)rect
 {
   [super drawRect:rect];
+  
   if (!_textStorage) {
     return;
   }
 
+#if TARGET_OS_OSX // [TODO(macOS GH#774)
+  return;
```

Comment:
This might be easy to miss when trying to understand this code. Basically the rest of this method is only used on iOS but it might not be obvious. @Saadnajmi I'm curious if you'd be ok with this sneaky `return` as-is.

Reply:
I noticed it. I haven't given too much thought to the actual code here quite yet, as I was prioritizing some of the other changes before this one :) 

_Technically_ , early returns are against our internal style guide ([aka.ms/objc-guide](https://microsoft.github.io/objc-guide/MethodsAndImplementations/EarlyReturns.html), but React Native macOS is kind of a special case. We generally prefer keeping diffs minimal and maybe this is the cleanest way to do so. I'll give it some thought :)

Reply:
Interesting. I don't mind early returns in general because they act like Swift guard statements, but perhaps the branching here would be cleaner if at the top of `setTextStorage:` we had:
```
#if TARGET_OS_OSX
  [self setMacTextStorage:...];
  return;
#endif
```

Reply:
Yeah, Swift guards are the case we _do_ have early returns :). Too bad there isn't Swift code in RN yet...

Reply:
Another option:
```
#if TARGET_OS_OSX
- (void)setTextStorage:(NSTextStorage *)textStorage
{
...
}
#else // iOS
- (void)setTextStorage:(NSTextStorage *)textStorage
{
...
}
```

Reply:
Ah.. this is a fun one because it also supports Catalyst in the implementation (line 267)..

It sounds like refactoring could cause unintended bugs (like https://github.com/microsoft/react-native-macos/pull/1346#discussion_r967386554 ?) so perhaps we keep it in the safe state? I think I'm ok with the early return so long as we document with a comment why we can skip the rest? If the issue was font smoothing, I think it's worth noting we do have our own fontsmoothing implementation here: https://github.com/microsoft/react-native-macos/pull/329 )